src/System Application/Test Library/User Details/src/UserDetailsTestLibrary.Codeunit.al

```diff
@@ -0,0 +1,113 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.TestLibraries.Security.User;
+
+using System.Security.User;
+
+/// <summary>
+/// Test library for the User Details module
+/// </summary>
+codeunit 132001 "User Details Test Library"
+{
+    var
+        UserDetailsRec: Record "User Details";
+        UserDoesNotExistErr: Label 'The user with security ID %1 does not exist', Locked = true;
+
+    /// <summary>
+    /// Saves the user details inside this instance, so that they can be access later.
+    /// </summary>
+    procedure FetchUserDetails()
```

Comment:
"GetUserDetails" to align with the interface function name. Also I would return the record to allow for more usages. I know it's public but if you both want to call functions in here and do something additional with the record..